theano/tensor/signal/downsample.py

```diff
@@ -208,9 +215,16 @@ def perform(self, node, inp, out):
         pc = zz.shape[-1]
         ds0, ds1 = self.ds
         st0, st1 = self.st
-        img_rows = x.shape[-2]
-        img_cols = x.shape[-1]
-
+        img_rows = x.shape[-2] + 2 * self.padding[0]
+        img_cols = x.shape[-1] + 2 * self.padding[1]
+        pad_h = self.padding[0]
+        pad_w = self.padding[1]
+
+        # pad the image
+        fill = x.min()-1
+        y = numpy.zeros((x.shape[0], x.shape[1], img_rows, img_cols)) + fill
+        y[:, :, pad_h:(y.shape[2]-pad_h), pad_w:(y.shape[3]-pad_w)] = x
```

Comment:
Why extract `y.shape[2]` and `y.shape[3]` here when you can just use `img_rows` and `img_cols`?